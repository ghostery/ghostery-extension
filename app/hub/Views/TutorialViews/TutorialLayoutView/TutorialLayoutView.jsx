/**
 * Tutorial Layout View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 *
 * ToDo: Update this file.
 */

import React from 'react';

/**
 * @class Implement the Tutorial Layout View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
const TutorialLayoutView = () => <div>Ghostery Hub - Tutorial Simple / Detailed Layout View</div>;

export default TutorialLayoutView;
