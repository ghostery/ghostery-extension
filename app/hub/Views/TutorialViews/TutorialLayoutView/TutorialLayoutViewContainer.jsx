/**
 * Tutorial Layout View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import TutorialLayoutView from './TutorialLayoutView';

/**
 * @class Implement the Tutorial Layout View for the Ghostery Hub
 * @extends Component
 * @memberof HubComponents
 */
class TutorialLayoutViewContainer extends Component {
	constructor(props) {
		super(props);

		// TODO call setTutorialNavigation action
		const { index } = this.props;
		this.props.actions.setTutorialNavigation({
			activeIndex: index,
			hrefPrev: `/tutorial/${index - 1}`,
			hrefNext: `/tutorial/${index + 1}`,
			hrefDone: '/',
			textPrev: t('hub_setup_nav_previous'),
			textNext: t('hub_setup_nav_next'),
			textDone: t('hub_setup_exit_flow'),
		});
	}

	render() {
		return <TutorialLayoutView />;
	}
}

// PropTypes ensure we pass required props of the correct type
TutorialLayoutViewContainer.propTypes = {
	index: PropTypes.number.isRequired,
	actions: PropTypes.shape({
		setTutorialNavigation: PropTypes.func.isRequired,
	}).isRequired,
};

export default TutorialLayoutViewContainer;
