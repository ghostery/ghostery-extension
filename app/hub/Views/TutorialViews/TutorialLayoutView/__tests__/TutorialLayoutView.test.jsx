/**
 * Tutorial Layout View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TutorialLayoutView from '../TutorialLayoutView';

describe('app/hub/Views/TutorialViews/TutorialLayoutView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('tutorial layout view is rendered correctly', () => {
			const component = renderer.create(<TutorialLayoutView />).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const component = shallow(<TutorialLayoutView />);
			expect(component.find('.TutorialLayoutView').length).toBe(1);
			expect(component.find('.TutorialLayoutView__image').length).toBe(2);
		});
	});
});
