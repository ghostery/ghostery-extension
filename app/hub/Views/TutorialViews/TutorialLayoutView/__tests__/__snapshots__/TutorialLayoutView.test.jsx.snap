// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/TutorialViews/TutorialLayoutView component Snapshot tests with react-test-renderer tutorial layout view is rendered correctly 1`] = `
<div
  className="TutorialLayoutView TutorialView--mediumFlexColumn row align-center-middle flex-container"
>
  <div
    className="columns small-10 medium-8 large-6 small-offset-1"
  >
    <img
      alt="hub_tutorial_simple_view"
      className="TutorialLayoutView__image layout-simple"
      src="/app/images/hub/tutorial/layout-simple.png"
    />
    <img
      alt="hub_tutorial_detailed_view"
      className="TutorialLayoutView__image layout-detailed"
      src="/app/images/hub/tutorial/layout-detailed.png"
    />
  </div>
  <div
    className="columns small-10 medium-6 large-4"
  >
    <div
      className="TutorialView__title"
    >
      hub_tutorial_layout_title
    </div>
    <div
      className="TutorialView__tagline"
    >
      hub_tutorial_layout_tagline
    </div>
  </div>
</div>
`;
